File: NHSOnline.Backend.Worker.IntegrationTests/Mocking/Models/Request.cs
using System.Collections.Generic;

namespace NHSOnline.Backend.Worker.IntegrationTests.Mocking.Models
{
    public class Request
    {

        public string UrlPath { get; set; }
        public string UrlPathPattern { get; set; }
        public string Method { get; set; }
        public readonly Dictionary<string, Dictionary<string, string>> Headers = new Dictionary<string, Dictionary<string, string>>();
        public readonly Dictionary<string, Dictionary<string, string>> QueryParameters = new Dictionary<string, Dictionary<string, string>>();
        public readonly IList<Dictionary<string, string>> BodyPatterns = new List<Dictionary<string, string>>();
    }
}

File: bddtests/src/main/kotlin/mocking/models/Request.kt
package mocking.models

class Request(@Suppress("unused")
              val url: String?,
              val urlPath: String?,
              val urlPattern: String?,
              val method: String,
              val headers: Map<String, Map<String, String>>,
              val queryParameters: Map<String, Map<String, String>>,
              val bodyPatterns: List<Map<String, String>>)
